### ISSUE TITLE
how to handle prefix?

### ISSUE THREAD

**Issue by JerryYangJin** [JuliaWeb/WebSockets.jl#65]
Is there anyway to specify and handle websocket `prefix`?

for example:
client side code: `ws = new WebSocket("ws://localhost:8080/websocket")`

how do I handle `prefix = /websocket` on julia websocket server?

Thank you!

**Comment by hustf** [comment]
Your 'websockethandle' function is called once per websocket you open. It receives two arguments: a request and websocket. You then enter a 'read' loop where only the websocket argument changes per message.

Your string "/websocket" exists somewhere in websocket.socket.buffer.data. You could dig this out from here and use it as an additional check before starting your read loop. Let me suggest a more flexible and faster way.

Websockets.jl has already done a handshake before this happens where it accepted the unspecified websocket protocol. You could instead specify your own protocol like this:

    ws = new WebSocket("ws://localhost:8080", "websocket-Jerry");

...but that won't work without extending WebSocket.jl's abilities, because "websocket-Jerry" was not returned as a confirmation. Fortunately, we are using Julia and can overload the function. The changes are at the bottom: "Sec-WebSocket-Protocol"

```
using WebSockets.
"""
Responds to a WebSocket handshake request.
Checks for required headers; sends Response(400) if they're missing or bad.
Otherwise, transforms client key into accept value, and sends Reponse(101).
Will reply with confirmation to any protocol.
"""
function websocket_handshake(request,client)
    if !haskey(request.headers, "Sec-WebSocket-Key")
        Base.write(client.sock, Response(400))
        return
    end
    if get(request.headers, "Sec-WebSocket-Version", "13") != "13"
        response = Response(400)
        response.headers["Sec-WebSocket-Version"] = "13"
        Base.write(client.sock, response)
        return
    end
    key = request.headers["Sec-WebSocket-Key"]
    if length(decode(Base64,key)) != 16 # Key must be 16 bytes
        Base.write(client.sock, Response(400))
        return
    end
  resp_key = generate_websocket_key(key)
  response = Response(101)
  response.headers["Upgrade"] = "websocket"
  response.headers["Connection"] = "Upgrade"
  response.headers["Sec-WebSocket-Accept"] = resp_key
  if haskey(request.headers, "Sec-WebSocket-Protocol") 
		response.headers["Sec-WebSocket-Protocol"] =  request.headers["Sec-WebSocket-Protocol"]
   		println("5")
  end 
  Base.write(client.sock, response)
end
```

Note that the modification above accepts *any* protocol directly. You can pick which 'read' loop to enter based on the request argument. But ideally, we would need to tell WebSockets which protocols we actually intend to implement on our server, and never accept unsupported protocols. That would mean WebSocket needs to expose more interface. Is that a good idea? @shashi ?
